etcd/v3/kv_etcd_test.go

```diff
@@ -150,6 +151,61 @@ func testWithRestarts(t *testing.T, testFn func(chan int, kvdb.Kvdb, *kvdb.KVPai
 
 }
 
+func TestKeys(t *testing.T) {
+	err := common.TestStart(true)
+	require.NoError(t, err, "Unable to start kvdb")
+	kv := newKv(t)
+	require.NotNil(t, kv)
+
+	data := []struct {
+		key, val string
+	}{
+		{"foo/keys/key1", "val1"},
+		{"foo/keys/key2", "val2"},
+		{"foo/keys/key3d/key3", "val3"},
+		{"foo/keys/key3d/key3b", "val3b"},
+	}
+	defer func() {
+		kv.DeleteTree("foo/keys")
+	}()
+
+	// load the data
+	for i, td := range data {
+		_, err := kv.Put(td.key, []byte(td.val), 0)
+		assert.NoError(t, err, "Unxpected error in Put on test# %d : %v", i+1, td)
+	}
+
+	// check the Keys()
+	results := []struct {
+		input    string
+		expected []string
+	}{
+		{"", []string{"foo"}},
+		{"non-existent", []string{}},
+		{"foo", []string{"keys"}},
+		{"foo/keys", []string{"key1", "key2", "key3d"}},
+		{"foo/keys/non-existent", []string{}},
```

Comment:
Can you add a test case where the input is `foo/keys/key3`? Even in that case it should also return `key3` & `key3B`

Reply:
Err.. it won't, the result will be the same as `foo/keys/non-existent`
* we're _not_ actually doing a "substring-search"  (e.g. searching for `foo/keys/key3*`)
* we'll add a separator to the end, and look for `foo/keys/key3/` -- so one has to provide a correct/existing prefix to get any results

But -- it's a good feedback (thanks!!) -- if this expectation was not obvious from the UT's, we _should_ make it obvious  (I'll expand the UTs).